beacon-chain/rpc/eth/beacon/handlers_pool.go

```diff
@@ -443,3 +446,138 @@ func (s *Server) ListBLSToExecutionChanges(w http.ResponseWriter, r *http.Reques
 		Data: changes,
 	})
 }
+
+// GetAttesterSlashings retrieves attester slashings known by the node but
+// not necessarily incorporated into any block.
+func (s *Server) GetAttesterSlashings(w http.ResponseWriter, r *http.Request) {
+	ctx, span := trace.StartSpan(r.Context(), "beacon.GetAttesterSlashings")
+	defer span.End()
+
+	headState, err := s.ChainInfoFetcher.HeadStateReadOnly(ctx)
+	if err != nil {
+		http2.HandleError(w, "Could not get head state: "+err.Error(), http.StatusInternalServerError)
+		return
+	}
+	sourceSlashings := s.SlashingsPool.PendingAttesterSlashings(ctx, headState, true /* return unlimited slashings */)
+	slashings := shared.AttesterSlashingsFromConsensus(sourceSlashings)
+
+	http2.WriteJson(w, &GetAttesterSlashingsResponse{Data: slashings})
+}
+
+// SubmitAttesterSlashing submits AttesterSlashing object to node's pool and
+// if passes validation node MUST broadcast it to network.
+func (s *Server) SubmitAttesterSlashing(w http.ResponseWriter, r *http.Request) {
+	ctx, span := trace.StartSpan(r.Context(), "beacon.SubmitAttesterSlashing")
+	defer span.End()
+
+	var req shared.AttesterSlashing
+	err := json.NewDecoder(r.Body).Decode(&req)
+	switch {
+	case err == io.EOF:
+		http2.HandleError(w, "No data submitted", http.StatusBadRequest)
+		return
+	case err != nil:
+		http2.HandleError(w, "Could not decode request body: "+err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	slashing, err := req.ToConsensus()
+	if err != nil {
+		http2.HandleError(w, "Could not convert request slashing to consensus slashing: "+err.Error(), http.StatusBadRequest)
+		return
+	}
+	headState, err := s.ChainInfoFetcher.HeadState(ctx)
```

Comment:
do you need a nil check on this service?

Reply:
Checking for nil for every field of the service would be a lot of boilerplate. I think we can safely assume that it's been initialized properly.